### DISCUSSION TITLE
Promote - how to know the current state of the deployment from the code?

### DISCUSSION THREAD

**Discussion by superfaz** [vercel/community#7353] [Help]
### Summary

Hi,
While using the standard promote process with custom domains (`deploy --prod --skip-domain` and then `promote`), I don't see a way to identify the state of the deployment in the code itself.
I don't see any change in the environment variables nor in the files available.

My use case: I am using Auth0 inside a NextJS project, which require the public url of the application (server side). To do so I use the VERCEL_URL environment variable. This is working for my tests before the promotion as the variable contains the domain name of the temporary deployment. After doing the `vercel promote`, the auth process fails because it tries to use the previous url and doesn't switch to the custom domain defined for the promotion.

I didn't find any changes inside the deployment after calling `vercel deploy`. Do you have any idea how to detect, from the code, if the deployment is staged or promoted?

### Steps to Reproduce

Create an app that displays VERCEL_URL for instance
Deploy using `vercel deploy --prod --skip-domain`
Check VERCEL_URL
Promote using `vercel promote`
Check VERCEL_URL (no change)

**Comment by pawlean** [comment]
Hi @superfaz, 

If I understood your issue correctly, here is my recommendation:

- The `VERCEL_URL` does not change. It remains as `<your-project>.vercel.app` [[1](https://vercel.com/docs/deployments/generated-urls)]
- If you want to change the URL to the custom domain, have you tried creating another environment variable with that domain and using that after running `vercel promote`?

**Reply by superfaz** [reply]
Hi @pawlean,

Thanks for your reply.

I agree with your recommendation and the standard environment variable `VERCEL_PROJECT_PRODUCTION_URL` contains the target domain for after the call to `vercel promote`.

My question is how to know, from the code, if the deployment is **staged** and use `VERCEL_URL` or if the deployment is **promoted** and use `VERCEL_PROJECT_PRODUCTION_URL`.

To say it an other way, I am trying to have a code such as the following working, but I have no clue about what should be the `TEST` part:

```ts
const baseUrl = TEST ? VERCEL_PROJECT_PRODUCTION_URL : VERCEL_URL;
```

[a comment by an automated account is not shown]
